### PR TITLE
[JENKINS-33414] Ensure that the tests do not leave the JVM marker flag set

### PR DESCRIPTION
**Pull request by stephenc**
See:

https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/4200/testReport/hudson.model/UpdateCenterTest/data/
https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/4200/testReport/hudson.model/DownloadServiceTest/testLoadJSON/
https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/4200/testReport/hudson.model/DownloadServiceTest/testLoadJSON/

@reviewbybees @jenkinsci/code-reviewers

### CODE REVIEW

**Comment by daniel-beck** [comment]
The last two links are the same URL.

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 1]
```diff
@@ -2890,6 +2893,9 @@ public void run() {
         LogFactory.releaseAll();
 
         theInstance = null;
+        if (JenkinsJVM.isJenkinsJVM()) {
+            JenkinsJVMAccess._setJenkinsJVM(oldJenkinsJVM);
```
Comment (jglick) [inline core/src/main/java/jenkins/model/Jenkins.java]: BTW is there really any use case for calling `_setJenkinsJVM(false)`? I mean, if the intended purpose of `JenkinsJVM` is to differentiate master from slave VMs, once a master always a master, regardless of whether it is shutting down, restarting, whatever. You potentially need a null check on `getInstance()` anyway, if your code might be run at weird times.

**Comment by jglick** [comment]
:bee:

**Comment by stephenc** [comment]
I'm going to go ahead and merge this as it is fixing tests that are failing

**Comment by KostyaSha** [comment]
:+1:

**Pull request merged**